lib/gtfs_realtime_viz.ex

```diff
@@ -148,26 +148,58 @@ defmodule GTFSRealtimeViz do
     end)
   end
 
-  @spec format_times([String.t] | nil) :: [String.t]
+  @spec format_times([{String.t, DateTime.t}] | nil) :: [Phoenix.HTML.Safe.t]
   def format_times(nil) do
     []
   end
   def format_times(time_list) do
     time_list
     |> Enum.sort()
     |> Enum.take(2)
-    |> Enum.map(& "#{Timex.format!(&1, "{h24}:{m}:{s}")}")
+    |> Enum.map(&format_time/1)
   end
 
-  @spec trainify([Proto.vehicle_position], Proto.vehicle_position_statuses, String.t) :: String.t
+  defp format_time({_, nil}) do
+    nil
+  end
+  defp format_time({trip_id, time}) do
+    ascii = Timex.format!(time, "{h24}:{m}:{s}")
+    span_for_id({ascii, trip_id})
+  end
+
+  @spec format_time_diff(time_list, time_list) :: [{time_output, time_output}]
+  when time_list: {String.t, DateTime.t} | nil, time_output: Phoenix.HTML.Safe.t | nil
+  def format_time_diff(base_list, nil) do
+    for format <- format_times(base_list) do
+      {format, nil}
+    end
+  end
+  def format_time_diff(nil, diff_list) do
+    for format <- format_times(diff_list) do
+      {nil, format}
+    end
+  end
+  def format_time_diff(base_list, diff_list) do
+    base_map = Map.new(base_list)
+    diff_map = Map.new(diff_list)
+    trips = for trip_id <- Map.keys(Map.merge(base_map, diff_map)),
+      base = base_map[trip_id],
+      diff = diff_map[trip_id],
+      is_nil(base) or is_nil(diff) or DateTime.compare(base, diff) != :eq do
```

Comment:
do we want to check for equality here? it seems likely that our predictions will almost always be slightly different here and maybe it would be good to check within a small threshold. On the other hand, i do think its nice to see how off we are even when its only a small amount.

Reply:
For now, showing the times if they're at all different is working for my use case. You all did the initial work for including TripUpdates: what were your goals?